hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFormatWriter.java

```diff
@@ -70,9 +72,10 @@
         this.output = fs.append(path, bufferSize);
       } catch (RemoteException e) {
         // this happens when either another task executor writing to this file died or
-        // data node is going down
-        if (e.getClassName().equals(AlreadyBeingCreatedException.class.getName())
-            && fs instanceof DistributedFileSystem) {
+        // data node is going down. Note that we can only try to recover lease for a DistributedFileSystem.
+        // ViewFileSystem unfortunately does not support this operation
+        if ((e.getClassName().contentEquals(AlreadyBeingCreatedException.class.getName()) || e.getClassName()
```

Comment:
suggest pull these conditions into a method named after the check its performing, for ease of reading.

Reply:
moved the recoverlease into a new method, not moving append handling since the next PR will make that code neater.